### DISCUSSION TITLE
use netty to implement java.nio.channels

### DISCUSSION THREAD

**Discussion by silence-coding** [netty/netty#12513] [Q&A]
I have a piece of software that relies on java.nio.channels and its underlying implementation uses junix socket. I was hoping to replace junix socket with netty. But I found that netty does not provide an adaptation layer for java.nio.channels. I want to use netty to implement java.nio.channels. Is there a library available?

**Comment by usertree** [comment]
Recently, it is found that Netty does not encapsulate the NIO interface. Can it support this capability?

**Comment by franz1981** [comment]
Nope, Netty is not just the raw socket implementation but it's a full fat networking event framework that requires users to adhere to a specific threading model to work at its best and it's not implementing the NIO api (instead, it uses it)

**Reply by silence-coding** [reply]
If you use netty to implement java.nio.channels, if not the best, it should be better than other implementations. We are currently looking for an implementation of the java unix domain channel, and now we find that the netty implementation may be better.
